Release: Update build release script to reflect latest changes

spinscale opened the pull request:
As the script now deploys to S3 and several things in master have
changed, this script needs to reflect the latest changes
- An unsigned RPM is built by default, so that users of older
  RPM based distros can download and use that RPM by default
- In addition a signed RPM is built, that is used for the repositories
- Paths for the new distributions have been fixed
- The check for the number of jars has been removed, as this is done
  as part of the license checking in `mvn verify`
- Checksum generation has been removed, as this is done as part of the
  mvn build
- Publishing artifacts of S3 has been removed
- Repostitory creation script has been updated

dadoonet commented:
I read the change and it makes sense to me. I did not test it though.

dadoonet commented on dev-tools/build_release.py:
```diff
@@ -358,63 +351,44 @@ def find_release_version(src_branch):
-def download_and_verify(release, files, plugins=None, base_url='https://download.elastic.co/elasticsearch/elasticsearch'):
+# Sample URL:
+# http://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/elasticsearch-rpm/2.0.0-beta1-SNAPSHOT/elasticsearch-rpm-2.0.0-beta1-SNAPSHOT.rpm
+def download_and_verify(release, files, plugins=None, base_url='https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution'):
```
Just a note here. We decided that by convention we will use the same naming as maven.
`groupId` has now changed to `org.elasticsearch.distribution.[packaging]` so I think we should also reflect that change here and use `org/elasticsearch/distribution/[packaging]` where `packaging` is:

* rpm
* deb
* zip
* tar

spinscale replied:
good  catch, fixed.

dadoonet commented:
I think it's good.

spinscale committed:
Release: Update build release script to reflect latest changes

As the script now deploys to S3 and several things in master have
changed, this script needs to reflect the latest changes

* An unsigned RPM is built by default, so that users of older
  RPM based distros can download and use that RPM by default
* In addition a signed RPM is built, that is used for the repositories
* Paths for the new distributions have been fixed
* The check for the number of jars has been removed, as this is done
  as part of the license checking in `mvn verify`
* Checksum generation has been removed, as this is done as part of the
  mvn build
* Publishing artifacts of S3 has been removed
* Repostitory creation script has been updated
